### PR TITLE
add support for synapse

### PR DESCRIPTION
**Pull request by albertobruin**

### CODE REVIEW

File: pkg/pipeline/pipeline.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ const (
 	AssetTypePostgresQuery        = AssetType("pg.sql")
 	AssetTypeRedshiftQuery        = AssetType("rs.sql")
 	AssetTypeMsSQLQuery           = AssetType("ms.sql")
+	AssetTypeSynapseQuery         = AssetType("sy.sql")
```
Comment (karakanb) [inline pkg/pipeline/pipeline.go:27]: I don't think `sy.sql` is obvious for synapse, shall we go with `synapse.sql`, @sabrikaragonen what are your thoughts?

Comment (sabrikaragonen) [reply pkg/pipeline/pipeline.go:27]: i agree, sy is very weird

File: pkg/synapse/materialization_test.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+			task: &pipeline.Asset{
+				Name: "my.asset",
+				Materialization: pipeline.Materialization{
+					Type: pipeline.MaterializationTypeView,
+				},
+			},
+			query: "SELECT 1",
+			want: []string{
+				"^DROP VIEW IF EXISTS my\\.asset;$",
+				"^CREATE VIEW my\\.asset AS SELECT 1;$",
```
Comment (sabrikaragonen) [inline pkg/synapse/materialization_test.go:37]: here a single statement would solve it I guess, CREATE OR ALTER VIEW

File: pkg/synapse/materialization_test.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			task: &pipeline.Asset{
+				Name: "my.asset",
+				Materialization: pipeline.Materialization{
+					Type: pipeline.MaterializationTypeTable,
+				},
+			},
+			query: "SELECT 1",
+			want: []string{"^SELECT tmp\\.\\* INTO #bruin_tmp_.+ FROM \\(SELECT 1\\) AS tmp;\n" +
+				"IF OBJECT_ID\\('my\\.asset', 'U'\\) IS NOT NULL DROP TABLE my\\.asset;\n" +
+				"SELECT \\* INTO my\\.asset FROM #bruin_tmp_.+;$"},
```
Comment (sabrikaragonen) [inline pkg/synapse/materialization_test.go:51]: deleting temp table after select into might be useful

File: pkg/synapse/materialization_test.go (hunk 2 above)
Comment (albertobruin) [reply pkg/synapse/materialization_test.go:37]: There is no `CREATE OR ALTER VIEW...` in synapse sql, despite it existing in ms sql

File: pkg/synapse/materialization_test.go (hunk 3 above)
Comment (albertobruin) [reply pkg/synapse/materialization_test.go:51]: Missed that. Thanks

**Pull request merged**